### PR TITLE
Fix several concurrency issues of analytics

### PR DESCRIPTION
**Pull request by duckofyork**
- Fix the bug where statistic method may fail when runing concurrently
- Fix the bug where degreeVertexProgram may fail when runing concurrently

### CODE REVIEW

File: grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/DegreeDistributionMapReduce.java [hunk 1]
```diff
@@ -47,14 +47,16 @@
     public DegreeDistributionMapReduce() {
     }
 
-    public DegreeDistributionMapReduce(Set<TypeId> selectedTypeIds) {
+    public DegreeDistributionMapReduce(Set<TypeId> selectedTypeIds, String degreeLabel) {
```
Comment (fppt) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/DegreeDistributionMapReduce.java]: For the sake of clarity `degreeLabel` should be a `TypeLabel`. Makes the method more readable and easier to work with form the outsider's perspective.

File: grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/MinMapReduce.java [hunk 2]
```diff
@@ -34,14 +34,14 @@
...
-public class MinMapReduce extends GraknMapReduce<Number> {
+public class MinMapReduce extends StatisticsMapReduce<Number> {
 
     // Needed internally for OLAP tasks
     public MinMapReduce() {
     }
 
-    public MinMapReduce(Set<TypeId> selectedTypeIds, String resourceDataType) {
-        super(selectedTypeIds, resourceDataType);
+    public MinMapReduce(Set<TypeId> selectedTypeIds, String resourceDataType, String degreeKey) {
```
Comment (fppt) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/MinMapReduce.java]: if `resourceDataType` does refer to the datatype of a resource type then this maybe should be `ResourceType.DataType dataType`. Again it just makes things more explicit.

Comment (duckofyork) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/MinMapReduce.java]: Done. 
ResourceType.DataType is not serialisable so I still need to use string inside the implementation method.

File: grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/DegreeDistributionMapReduce.java (hunk 1 above)
Comment (duckofyork) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/DegreeDistributionMapReduce.java]: Degree is a property on the vertex. So I guess I could change the variable name to sth like degreePropertyKey?

**Pull request closed without merging**